hcl/hcl.go

```diff
@@ -873,16 +879,135 @@ func CopyBody(target *hclwrite.Body, src *hclsyntax.Body, eval PartialEvaluator)
 	logger.Trace().Msg("Append blocks.")
 
 	for _, block := range src.Blocks {
-		targetBlock := target.AppendNewBlock(block.Type, block.Labels)
-		if block.Body == nil {
+		err := appendBlock(target, block, eval)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func appendBlock(target *hclwrite.Body, block *hclsyntax.Block, eval Evaluator) error {
+	if block.Type == "tm_dynamic" {
+		return appendDynamicBlock(target, block, eval)
+	}
+
+	targetBlock := target.AppendNewBlock(block.Type, block.Labels)
+	if block.Body != nil {
+		err := CopyBody(targetBlock.Body(), block.Body, eval)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func appendDynamicBlock(target *hclwrite.Body, block *hclsyntax.Block, eval Evaluator) error {
+	attrs := block.Body.Attributes
+
+	errs := errors.L()
+	if len(block.Labels) != 1 {
+		errs.Append(errors.E(ErrTerramateSchema,
+			block.LabelRanges, "tm_dynamic requires a label"))
+	}
+	genBlockType := block.Labels[0]
```

Comment:
hmm not returning from the function if labels is 0 and executing this will panic, so for an invalid config we would panic.

Reply:
thanks!

error list is always hard for me because my brain always thinks we return on the first error...